src/Compilers/Core/Portable/MetadataReader/TypeNameDecoder.cs

```diff
@@ -181,7 +191,7 @@ internal TypeSymbol GetTypeSymbol(MetadataHelpers.AssemblyQualifiedTypeName full
             if (fullName.TypeArguments != null)
             {
                 ImmutableArray<bool> argumentRefersToNoPiaLocalType;
-                var typeArguments = ResolveTypeArguments(fullName.TypeArguments, out argumentRefersToNoPiaLocalType);
+                var typeArguments = ResolveTypeArguments(fullName.TypeArguments, out argumentRefersToNoPiaLocalType, allowTypeParameters);
```

Comment:
It looks like we decode `!n` and `!!n` only when they are in type argument position. Can they be also array element type for example (`!1[]`)?

Reply:
Good catch. Thanks